utils/rgmercs_utils.lua

```diff
@@ -861,12 +861,16 @@ function Utils.UseSong(songName, targetId, bAllowMem)
             mq.doevents()
             mq.delay(1)
         end
-
+        
         if targetId == me.ID() then
-            mq.delay("5s", function() return me.Song(songName).Duration.Seconds() > oldDuration end)
+            mq.delay("3s", function() return me.Song(songName).Duration.Seconds() > oldDuration end)
+        else
+            -- bard songs take a bit to refresh after casting window closes, otherwise we'll clip our song
+            -- might need to be bumped a bit for people with high latency?
+            mq.delay(150)

```

Comment:
Should this be longer with a early out if Casting.ID() > 0?

Reply:
Yep that sounds like a much more stable check than hoping to pick something long enough to accommodate every latency

Reply:
Actually I think that sole check can take care of both situations -- no need for the oldDuration either, right?  There's no difference in how bard songs are cast no matter who the target is, we only care that they've finished singing

Reply:
yah